### PR TITLE
[Blueprint] Add support for change notes; other code improvements

### PR DESCRIPTION
**Pull request by filizt**
## Description

This PR has the following changes:
- Update Blueprint SDK version
- Add ability to include change notes during publish operation
- Import operation should register RP since it's creating a PUT call
- Template artifact parameter values can be strings or a collection. Add support for both.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ImportAzureRMBlueprint.cs [hunk 1]
```diff
@@ -49,6 +49,8 @@ public class ImportAzureRmBlueprint : BlueprintDefinitionCmdletBase
         #region Cmdlet Overrides
         public override void ExecuteCmdlet()
         {
+            RegisterBlueprintRp(SubscriptionId ?? DefaultContext.Subscription.Id);
```
Comment (cormacpayne) [inline src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ImportAzureRMBlueprint.cs]: @filizt why is this needed?

File: src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/PublishAzureRMBlueprint.cs [hunk 2]
```diff
@@ -31,10 +34,10 @@ public class PublishAzureRmBlueprint : BlueprintDefinitionCmdletBase
         [ValidateNotNullOrEmpty]
         public string Version { get; set; }
 
-        // To-Do: ChangeNotes will be added in the next SDK release
-        /* [Parameter(ParameterSetName = ParameterSetNames.PublishBlueprint, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "To-Do")]
+        [Parameter(ParameterSetName = ParameterSetNames.PublishBlueprint, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ChangeNotes)]
+        [ValidateLength(0,500)]
         [ValidateNotNullOrEmpty]
-        public string ChangeNotes { get; set; }*/
+        public string ChangeNotes { get; set; }
```
Comment (cormacpayne) [inline src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/PublishAzureRMBlueprint.cs]: @filizt since we need to avoid plural parameter names in PowerShell, we should rename this to something like `-ChangeContent` or `-ChangeRecord` (since `-ChangeNote` doesn't really make sense).

Also, since we're adding a new parameter, you will want to regenerate the markdown help for this cmdlet.

File: src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ImportAzureRMBlueprint.cs (hunk 1 above)
Comment (filizt) [reply src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ImportAzureRMBlueprint.cs]: @cormacpayne import cmdlet makes a PUT call to create a blueprint and it could be the case that user hasn't registered the RP with the subscription yet. This call is making sure Blueprint service is registered.

File: src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/PublishAzureRMBlueprint.cs (hunk 2 above)
Comment (filizt) [reply src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/PublishAzureRMBlueprint.cs]: I ended up making it ChangeNote since everywhere else we're calling it ChangeNotes and not calling it something else would not make senses. Let me know if this works.

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/psnetsign-esrp/442/
